sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/HttpResponse.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk;
+
+
+import kalix.javasdk.impl.ByteContent;
+import kalix.javasdk.impl.JsonContent;
+import kalix.javasdk.impl.NoContent;
+
+import javax.annotation.Nullable;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public interface HttpResponse {
+  StatusCode.Success getStatusCode();
+
+  @Nullable
+  String getContentType();
+
+  byte[] getBody();
```

Comment:
We can have a convenient method to get the body deserialized to a given type. Only for usage with Json payloads. 

Something like: 

```
public <T> T bodyAs(Class<T> clz) {...}
```

I have played a bit with an integration test returning HttpResponse and it's quite awkward because we need to take the byte array, go look for a ObjectMapper somewhere (I know where to find one, but users have no clue) and then deserialize the payload. 

We have already static methods to access our internal ObjectMapper so we don't really need to depend on an instance of JsonSupport.

Users can always decide to call `getBytes` and deserialize it the way they want if they are not interested in using our serializers.

Reply:
Ok, check now. There is a direct dependency on JsonSupport, but I don't see any difference with JsonSuport vs ObjectMapper dependency.